File: lab4.py
import math
Exit = 0
spisok =[]
maxS=0
minS=0
while Exit == 0:
    i = 0
    print("Ввести a")
    a = float(input())
    print("Ввести нижнюю границу - x")
    x1 = float(input())
    print("Ввести вверхнюю границу - x")
    x2 = float(input())
    print("Ввести x")
    x = float(input())
    print('Ввести шаг изменения переменной x')
    Step = float(input())
    print("Выбрать номер функции 1-3")
    ch = int(input())
    if ch == 1:
        while x1 <= x and x <= x2:
            g1 = (3*a*a-7*a*x+4*x*x)
            if g1 != 0:
                g=(-7)*(20*a*a+11*a*x+45*x*x)/(3*a*a-7*a*x+4*x*x)
                spisok.append(g)
                print('Аргумент x = {:2f}\t Значение = {:2f}'.format(x, spisok[i]))
                i = i+1
                x = x + Step
                maxS = max(spisok)
                minS = min(spisok)
            else:
                print("Значения не принадлежат области определения функции.")
                x = x1-1
                continue

    elif ch == 2:
        while x1 <= x and x <= x2:
            f1 = (60 * (a**2) + 88* (a*x) +21 * (x**2))
            if f1 != -1 and f1 != 1:
                f = math.tan(60*a*a+88*a*x-21*x*x)
                spisok.append(f)
                print('Аргумент x = {:2f}\t Значение = {:2f}'.format(x, spisok[i]))
                x = x + Step
                i = i+1
                maxS = max(spisok)
                minS = min(spisok)
            else:
                print("Значения не принадлежат области определения функции.")
                x=x1-1
                continue
    elif ch == 3:
        while x1 <= x and x <= x2:
            y1 = (-40) * a * a + 3 * a * x + x * x + 1
            if y1 >= 0:
                Y = math.log(y1)
                y=math.log((-40) * a * a + 3 * a * x + x * x + 1)/math.log(2)
                spisok.append(y)
                print('Аргумент x = {:2f}\t Значение = {:2f}'.format(x, spisok[i]))
                x = x + Step
                i = i+1
                maxS = max(spisok)
                minS = min(spisok)
            else:
                print("Значения не принадлежат области определения функции.")
                x = x1-1
                continue
    else:
        print("Нет такой функции.")
    print('Максимум'+str(maxS))
    print('Минимум'+str(minS))
    print("Выйти из программы?(1-да, 0-нет)")
    Exit = int(input())

File: lab4.c
#include <stdio.h>
#include <math.h>

int main(void)
{
float a, x, g, f, y, step, x1, H, mini, maxi;
int var, exitt, n, steps;
float spisok[100];
exitt = 0;
while (exitt == 0)
{
    n = 0;
    printf ("Vvedite a: ");
    scanf ("%f" , &a);
    printf ("Vvedite nomer: ");
    scanf ("%int", &var);
    printf ("Vvedite na4al'noe x:  ");
    scanf ("%f", &x);
    printf ("Vvedite kone4noe x:  ");
    scanf ("%f", &x1);
    printf ("Vvedite koli4estvo shagov  ");
    scanf ("%i", &steps);
    if (x>x1)
    {
        step = (-(x-x1)/steps);
    }
    else
    {
        step = ((x1-x)/steps);
    }
    if (var== 1)
    {
            if (3 * (a*a) - 7 * a * x + 4 * x * x == 0)
            {
                printf("Nevernoe zna4enie\n");
            }
            else
            {
                while (x != x1)
                    {
                        g = -((7) * (20 *(a*a) + 11 * a * x - ( 45 * (x*x)))) / (3 *(a*a) - 7 * a * x + 4*x*x);
                        printf("g=%f\n",g);
                        x = x+step;
                        spisok[n] =g;
                        n++;
                        if (mini > g)
                        {
                            mini = g;
                        }
                        if (maxi < g)
                        {
                            maxi = g;
                        }
                    }
            }
    }
    if (var == 2)
    {
            H = 60 * (a*a) + 88* (a*x) +21 * (x*x);
            if (H!=1 && H!=-1)
            {
                while (x != x1)
                {      
                f = (tan(60 * (a *a) + 88 * (a * x) + 21 * (x*x)));
                printf("f=%f\n",f);
                x = x + step;
                spisok[n] =g;
                n++;
                if (mini > g)
                {
                    mini = g;
                }
                if (maxi < g)
                {
                    maxi = g;
                }
                }
            }
            else
            {
                printf("Nevernoe zna4enie\n");
            }
        }
    if (var == 3)
    {
            if ((-40) * (a*a) + 3* a* x+ (x*x) +1 > 0)
            {
                while (x != x1)
                {
                y = log((-40) * (a*a) + 3* a* x+ (x*x) +1)/log(2);
                printf("y=%f\n",y);
                x = x+step;
                spisok[n] =g;
                n++;
                if (mini > g)
                {
                    mini = g;
                }
                if (maxi < g)
                {
                    maxi = g;
                }
                }
            }
            else
            {
                printf("Nevernoe zna4enie\n");
            }
        }
    printf("max = %f ", maxi);
    printf("min = %f\n ", mini);
    if (var > 3 || var < 1)
    {
        printf("Nevernoe zna4enie\n");
    }
    printf("Prodol)|(ut' raboty? (0-Yes, 1-Net)");
    scanf("%int", &exitt);
}
return 0;
}
